File: ProjectEuler_FSharp/Palindromes.fs
module ProjectEuler_FSharp.Palindromes

open System

let private reverse value =
    let rec reverse_iter(reversed: int, remain: int) =
        if remain > 0 then
            reverse_iter(reversed * 10 + remain % 10, remain / 10)
        else
            reversed
    reverse_iter(0, value)

let IsPalindrome value = 
    value = reverse value

File: ProjectEuler_CSharp/Palindromes.cs
namespace ProjectEuler_CSharp
{
    public static class Palindromes
    {
        public static bool IsPalindrome(int value) => value == reverse(value);

        private static int reverse(int x)
        {
            int remain = x, reversed = 0;
            while(remain > 0) {
                reversed = reversed * 10 + remain % 10;
                remain = remain / 10;
            }
            return reversed;
        }
    }
}
